Documentation/deploy-master.md

```diff
@@ -301,6 +286,86 @@ spec:
       timeoutSeconds: 1
 ```
 
+### Run the per host calico agent
+The per host Calico agent runs on all hosts including the master node. It is responsible for enforcing networking policy and connecting the pods on that host to the flannel overlay network.
+
+When creating `/etc/systemd/system/calico-node.service`:
+
+* Replace `${ETCD_ENDPOINTS}`
+* Replace `${ADVERTISE_IP}` with this node's publicly routable IP.
+
+**/etc/systemd/system/calico-node.service**
+
+```yaml
+[Unit]
+Description=Calico per-host agent
+Requires=network-online.target
+After=network-online.target
+
+[Service]
+Slice=machine.slice
+Environment=CALICO_DISABLE_FILE_LOGGING=true
+Environment=HOSTNAME=${ADVERTISE_IP}
+Environment=IP=${ADVERTISE_IP}
+Environment=FELIX_FELIXHOSTNAME=${ADVERTISE_IP}
+Environment=CALICO_NETWORKING=false
+Environment=NO_DEFAULT_POOLS=true
+Environment=ETCD_ENDPOINTS=${ETCD_ENDPOINTS}
+ExecStart=/usr/bin/rkt run --inherit-env --stage1-from-dir=stage1-fly.aci \
+--volume=modules,kind=host,source=/lib/modules,readOnly=false \
+--mount=volume=modules,target=/lib/modules \
+--trust-keys-from-https quay.io/calico/node:v0.19.0
+
+KillMode=mixed
+Restart=always
+TimeoutStartSec=0
+
+[Install]
+WantedBy=multi-user.target
+```
+
+### Set Up the policy-agent Pod
+
+The policy agent is the last major piece of the our master node. It monitors the API for changes related to network policy and configures Calico to implement that policy.
+
+When creating `/etc/kubernetes/manifests/policy-agent.yaml`:
+
+* Replace `${ETCD_ENDPOINTS}`
+
+**/etc/kubernetes/manifests/policy-agent.yaml**
+
+```yaml
+apiVersion: v1
+kind: Pod 
+metadata:
+  name: calico-policy-agent
+  namespace: calico-system 
+spec:
+  hostNetwork: true
```

Comment:
I agree with @chom on this.

Couple clarifying points:

The ETCD_ENDPOINTS aren't on the pod network itself (unless this has changed elsewhere), but they are routable from any node in the cluster.

I'm not sure if the policy-agent knows how to securely speak with the api-server (but ideally it should). The two main options developers generally provide are getting the necessary data from a kubeconfig file, or using "InClusterConfig" (discovers api-server from env var in the pod and service account data mounted at known location).